File: attachment-microservice/src/test/scala/TestAttachmentController.scala
import java.io.File
import java.net.URI

import org.junit.{Assert, Test}
import org.springframework.core.io.FileSystemResource
import org.springframework.util.LinkedMultiValueMap
import org.springframework.web.client.RestTemplate

/**
  * @author 刘佳兴
  * @version V1.0
  */
class TestAttachmentController {

  @Test
  def testUpload(): Unit = {
    val restTemplate = new RestTemplate()
    val uri = new URI("http://localhost:9000/upload")
    val resource = new FileSystemResource("/home/liujx/Pictures/Selection_003.png")
    val multiValueMap = new LinkedMultiValueMap[String, Object]()
    multiValueMap.add("file", resource)
    multiValueMap.add("relativePath", "/")

    val result = restTemplate.postForObject(uri, multiValueMap, classOf[String])
    Assert.assertNotNull(result)
  }

  @Test
  def testReadFile: Unit = {
    val restTemplate = new RestTemplate()
    val uri = new URI("http://localhost:9000/read/Selection_003.png")

    val result = restTemplate.getForObject(uri, classOf[File])
    println(result)
  }

}

File: attachment-microservice/src/test/java/TestAttachmentController.java
import org.junit.Assert;
import org.junit.Test;
import org.springframework.core.io.FileSystemResource;
import org.springframework.util.LinkedMultiValueMap;
import org.springframework.web.client.RestTemplate;

import java.net.URI;
import java.net.URISyntaxException;

/**
 * @author 刘佳兴
 */
public class TestAttachmentController {

    @Test
    public void testUpload() throws URISyntaxException {
        RestTemplate restTemplate = new RestTemplate();
        URI uri = new URI("http://localhost:9000/upload/user.image.a.b.c.d");

        FileSystemResource resource = new FileSystemResource("/home/liujx/Pictures/Selection_003.png");
        LinkedMultiValueMap<String, Object> multiValueMap = new LinkedMultiValueMap<>();
        multiValueMap.add("file", resource);

        String result = restTemplate.postForObject(uri, multiValueMap, String.class);

        System.out.println(result);

        Assert.assertNotNull(result);
    }

}
